modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -544,6 +524,35 @@ public void onShutdown() {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public CompletableFuture<Void> onBeforeApply(Command command) {
+        if (command instanceof SingleKeyCommand) {
```

Comment:
What if given command in non-transactional?

Reply:
This step will be no-op